echion/coremodule.cc

```diff
@@ -374,6 +375,106 @@ static PyObject* init_asyncio(PyObject* Py_UNUSED(m), PyObject* args)
     Py_RETURN_NONE;
 }
 
+// ----------------------------------------------------------------------------
+static PyObject* track_greenlet(PyObject* Py_UNUSED(m), PyObject* args)
+{
+    uintptr_t greenlet_id;  // map key
+    PyObject* name;
+    PyObject* frame;
+
+    if (!PyArg_ParseTuple(args, "lOO", &greenlet_id, &name, &frame))
+        return NULL;
+
+    StringTable::Key greenlet_name;
+
+    try
+    {
+        greenlet_name = string_table.key(name);
```

Comment:
I think this is ok in Echion, but probably not when we write and similar wrapper for Stack v2. 

The caller of `track_greenlet()` in dd-trace-py would be the `MainThread`, and `string_table` is also modified and accessed by the sampler thread, which a separate `std::thread`, right?

Reply:
ok, but don't think it's safe to access/modify the underlying `unordered_map` with multiple threads...

Reply:
Ah right. Perhaps we can fix this in the string table instead.

Reply:
Thanks for a quick fix! 
When we upgrade echion on dd-trace-py, I'll make sure to keep an eye on perf impact of these all together.